pkg/services/cleanup/cleanup.go

```diff
@@ -57,66 +62,97 @@ type CleanUpService struct {
 	tempUserService           tempuser.Service
 }
 
+type cleanUpJob struct {
+	name string
+	fn   func(context.Context, log.Logger)
+}
+
+func (j cleanUpJob) String() string {
+	return strconv.Quote(j.name)
+}
+
 func (srv *CleanUpService) Run(ctx context.Context) error {
-	srv.cleanUpTmpFiles()
+	srv.cleanUpTmpFiles(ctx, srv.log)
 
 	ticker := time.NewTicker(time.Minute * 10)
 	for {
 		select {
 		case <-ticker.C:
-			ctxWithTimeout, cancelFn := context.WithTimeout(ctx, time.Minute*9)
-			defer cancelFn()
-
-			srv.cleanUpTmpFiles()
-			srv.deleteExpiredSnapshots(ctx)
-			srv.deleteExpiredDashboardVersions(ctx)
-			srv.deleteExpiredImages(ctx)
-			srv.cleanUpOldAnnotations(ctxWithTimeout)
-			srv.expireOldUserInvites(ctx)
-			srv.deleteStaleShortURLs(ctx)
-			srv.deleteStaleQueryHistory(ctx)
-			err := srv.ServerLockService.LockAndExecute(ctx, "delete old login attempts",
-				time.Minute*10, func(context.Context) {
-					srv.deleteOldLoginAttempts(ctx)
-				})
-			if err != nil {
-				srv.log.Error("failed to lock and execute cleanup of old login attempts", "error", err)
-			}
+			srv.clean(ctx)
 		case <-ctx.Done():
 			return ctx.Err()
 		}
 	}
 }
 
-func (srv *CleanUpService) cleanUpOldAnnotations(ctx context.Context) {
+func (srv *CleanUpService) clean(ctx context.Context) {
+	const timeout = 9 * time.Minute
+	start := time.Now()
+	ctx, span := srv.tracer.Start(ctx, "cleanup background job")
+	defer span.End()
+	tID := tracing.TraceIDFromContext(ctx, false)
+	logger := srv.log.New("traceID", tID)
+	ctx, cancelFn := context.WithTimeout(ctx, timeout)
+	defer cancelFn()
+
+	cleanupJobs := []cleanUpJob{
+		{"clean up temporary files", srv.cleanUpTmpFiles},
+		{"delete expired snapshots", srv.deleteExpiredSnapshots},
+		{"delete expired dashboard versions", srv.deleteExpiredDashboardVersions},
+		{"delete expired images", srv.deleteExpiredImages},
+		{"cleanup old annotations", srv.cleanUpOldAnnotations},
+		{"expire old user invites", srv.expireOldUserInvites},
+		{"delete stale short URLs", srv.deleteStaleShortURLs},
+		{"delete stale query history", srv.deleteStaleQueryHistory},
+		{"delete old login attempts", srv.deleteOldLoginAttempts},
+	}
+
+	logger.Debug("Starting cleanup jobs", "jobs", fmt.Sprintf("%v", cleanupJobs))
```

Comment:
Here's another nitpick, but I think this could be easier to parse. What do you think about formatting with something like `strings.Join` instead? 

The current output isn't bad though: 
```
DEBUG[09-20|09:52:36] Starting cleanup jobs                    logger=cleanup traceID=00000000000000000000000000000000 jobs="[\"clean up temporary files\" \"delete expired snapshots\" \"delete expired dashboard versions\" \"delete expired images\" \"cleanup old annotations\" \"expire old user invites\" \"delete stale short URLs\" \"delete stale query history\" \"delete old login attempts\"]"
```